### PR TITLE
Add a technical review requirement for feature documentation

### PR DESCRIPTION
**Pull request by annajung**
- Adds technical review requirement for feature documentation
- Modify to spells out k in `k/k` and `k/enhancement`

### CODE REVIEW

File: content/en/docs/contribute/new-content/new-features.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ deadlines.
 1. Open a pull request against the
 `dev-{{< skew nextMinorVersion >}}` branch in the `kubernetes/website` repository, with a small
 commit that you will amend later.
-2. Edit the pull request description to include links to `k/k` PR(s) and `k/enhancement` issue(s).
+2. Edit the pull request description to include links to `kubernetes/kubernetes` PR(s) and `kubernetes/enhancement` issue(s).
```
Comment (sftim) [inline content/en/docs/contribute/new-content/new-features.md:101]: Those repo names could be hyperlinks, if you like.

Comment (sftim) [reply content/en/docs/contribute/new-content/new-features.md:101]: _k/enhancement_ is actually https://github.com/kubernetes/enhancements (note plural)

**Review by sftim — COMMENTED** [review]
/lgtm

File: content/en/docs/contribute/new-content/new-features.md (hunk 1 above)
Comment (annajung) [reply content/en/docs/contribute/new-content/new-features.md]: Updated! Thanks for catching that :D

**Comment by tengqm** [comment]
/lgtm
/approve
I believe @sftim's lgtm still stands.

**Pull request merged**